pyanaconda/modules/subscription/runtime.py

```diff
@@ -15,14 +15,67 @@
 # License and may only be used or replicated with the express permission of
 # Red Hat, Inc.
 #
+import os
+import json
+
 from dasbus.typing import get_variant, Str
+from dasbus.connection import AddressedMessageBus
+from dasbus.error import DBusError
+
+from pyanaconda.modules.common.constants.objects import RHSM_REGISTER
+
+from pyanaconda.core.i18n import _
 
 from pyanaconda.modules.common.task import Task
+from pyanaconda.modules.common.constants.services import RHSM
+from pyanaconda.modules.common.constants.objects import RHSM_REGISTER
 
 from pyanaconda.anaconda_loggers import get_module_logger
 log = get_module_logger(__name__)
 
 
+def start_rhsm_private_bus(rhsm_register_server_proxy, message_bus_class=AddressedMessageBus):
+    """Start the RHSM private DBus.
+
+    The RHSM private DBus session is used for secure credential passing.
+
+    NOTE: There is no locking involved, in starting/stopping of
+          the private bus. So it's important to make sure only
+          one task using the RHSM private bus is running at a time.
+          Otherwise unexpected behavior might happen (one task shutting down
+          private bus another task is still using, etc.).
+
+    :param rhsm_register_server_proxy: DBus proxy for the RHSM RegisterServer object
+    :param message_bus: suitable message bus class, such as AddressedMessageBus
+    :return: address of the RHSM private DBus session
+    :rtype: str
+    """
+    log.debug("subscription: starting RHSM private DBus session")
+    locale = os.environ.get("LANG", "")
+    private_bus_address = rhsm_register_server_proxy.Start(locale)
+    private_connection = message_bus_class(private_bus_address)
+    private_register_proxy = private_connection.get_proxy(RHSM.service_name,
+                                                          RHSM_REGISTER.object_path)
+    log.debug("subscription: RHSM private DBus session started")
+    return private_register_proxy
+
+
+def stop_rhsm_private_bus(rhsm_register_server_proxy):
+    """Stop the RHSM private DBus.
+
+    The RHSM private DBus session is used for secure credential passing.
+    We are only enabling the private bus for credential passing and
+    shut it down once the credentials have been transferred.
+
+    :param rhsm_register_server_proxy: DBus proxy for the RHSM RegisterServer object
+    """
+
+    log.debug("subscription: shutting down the RHSM private DBus session")
+    locale = os.environ.get("LANG", "")
+    rhsm_register_server_proxy.Stop(locale)
```

Comment:
I would recommend to call the `disconnect` method of the message bus if possible. It is not really necessary, just a nice thing to do.

Reply:
Good point, added.